service/resource/namespace/resource.go

```diff
@@ -88,6 +89,17 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		}
 	}
 
+	// In case the namespace is already terminating we do not need to do any
+	// further work. Then we cancel the reconciliation to prevent the current and
+	// any further resource from being processed.
+	if namespace.Status.Phase == "Terminating" {
+		canceler, exists := cancelercontext.FromContext(ctx)
+		if exists {
+			canceler <- struct{}{}
+			return nil, nil
+		}
+	}
```

Comment:
This is all of the magic we need to do here to be able to cancel the resource processing. Note that I actually wanted to use a constant for `Terminating` as defined in the k8s client libs, bit the api core package had strange issues with proto buf magic while vendoring so I just went with the plain string.